scipy/optimize/_constraints.py

```diff
@@ -166,23 +166,33 @@ class Bounds:
 
     Parameters
     ----------
-    lb, ub : array_like
+    lb, ub : array_like, optional
         Lower and upper bounds on independent variables. Each array must
         have the same size as x or be a scalar, in which case a bound will be
         the same for all the variables. Set components of `lb` and `ub` equal
         to fix a variable. Use ``np.inf`` with an appropriate sign to disable
         bounds on all or some variables. Note that you can mix constraints of
         different types: interval, one-sided or equality, by setting different
-        components of `lb` and `ub` as necessary.
+        components of `lb` and `ub` as necessary. Defaults to lb = 0 and
+        ub = np.inf (non-negativity).
     keep_feasible : array_like of bool, optional
         Whether to keep the constraint components feasible throughout
         iterations. A single value set this property for all components.
         Default is False. Has no effect for equality constraints.
     """
-    def __init__(self, lb, ub, keep_feasible=False):
-        self.lb = np.asarray(lb)
-        self.ub = np.asarray(ub)
-        self.keep_feasible = keep_feasible
+    def _input_validation(self):
+        try:
+            res = np.broadcast_arrays(self.lb, self.ub, self.keep_feasible)
+            self.lb, self.ub, self.keep_feasible = res
+        except:
+            message = "`lb` and `ub` must have the same shape."
+            raise ValueError(message)
+
+    def __init__(self, lb=0, ub=np.inf, keep_feasible=False):
```

Comment:
From my point of view (working with non-linear minimisation) it makes much more sense to have `lb=-np.inf`.

Reply:
Can we also make it work with `{lb, ub}=None`, which would correspond to `{lb, ub}=-np.inf/np.inf`?

Reply:
We could. That would be totally fine if users would only enter literally `lb=None` (which would become `-np.inf` internally) and `ub=None` (which would become `np.inf` internally). I'm concerned, though, that users will use `None` as an element of a `lb`/`ub` sequence, and these would turn into object arrays. We could replace the `None`s with +/- `np.inf` and change the type, but is this what you intended?